MekHQ/src/mekhq/campaign/ResolveScenarioTracker.java

```diff
@@ -1160,18 +1153,18 @@ public void resolveScenario(int resolution, String report) {
             }
         }
         // update prisoners
-        for(UUID pid : prisonerStatus.keySet()) {
+        for (UUID pid : prisonerStatus.keySet()) {
             PrisonerStatus status = prisonerStatus.get(pid);
             Person person = status.getPerson();
-            if(null == person || null == status) {
+            if (null == person) {
                 continue;
             }
             MekHQ.triggerEvent(new PersonBattleFinishedEvent(person, status));
-            if(status.isDead()) {
+            if (status.isDead()) {
                 continue;
             }
             if (status.isCaptured()) {
-                getCampaign().recruitPerson(person, true, true);
+                getCampaign().recruitPerson(person, true, false, false, true);
```

Comment:
I think this PR is fine, but it is a shame Java doesn't have named parameters to make booleans like this easier to read.

The alternative would be to use Enumerations.

Reply:
Yeah, thankfully Idea shows what each mean when you use true/false variables